### DISCUSSION TITLE
How to get the status of the last command?

### DISCUSSION THREAD

**Discussion by oliwer** [wryun/es-shell#47] [Q&A] [answered]
Hi there.

When using the shell interactively, I like to be notified when a command fails. In `rc` this is easy to do: I just need to check if `$status` contains an element !~ 0. But how can I do the same with `es`? Obviously I cannot use `<={...}`.

**Comment by mwgamera** [comment, marked as answer]
It's not as obvious as it should be, but putting something along the lines of the following into ~/.esrc should save the result of each command as the global $status variable in login shells:
```
noexport = $noexport status
fn %eval-noprint { status = <=$* }
fn %eval-print { echo $* >[1=2]; status = <=$* }
```

I you want it in non-login interactive shell, then I'm afraid it might require modifying initial.es and recompiling. (cf. #33)

**Reply by oliwer** [reply]
Thanks! That's good enough for my need. It was indeed not easy to guess from the manpage.
